### ISSUE TITLE
[Beta] User list doesn't appear for sharing

### ISSUE THREAD

**Issue by cwmke** [owncloud/android#1386]
Using fdroid's owncloud beta151222 on a Nexus 6p with PureNexus rom, I perform the following steps,

1. Long press on a folder I would like to share with another user.
2. Press the share button in the top bar.
3. Click 'Add User Or Group'.
4. Type in user's name.
5. No list pops up and pressing the enter button does nothing.

I noticed it still works well in the stable version.  This essentially disables sharing with other users.  Thanks for any help you can provide.  I'm really enjoying beta.

**Comment by tobiasKaminsky** [comment]
This is strange as this feature is already in the official version and therefore it is not beta specific anymore. I include the master branch every time I update the beta and for me it is working.

Can you try to provide a log? (I think there is a new navigation entry "log" in the navigation drawer)

**Comment by tobiasKaminsky** [comment]
@davivel Maybe you have an idea why this happens?

**Comment by cwmke** [comment]
Here's the logcat errors.  The in-app log was blank.
````
12-25 10:35:25.913 W/ActivityManager(6181): Permission Denial: opening provider com.owncloud.android.providers.UsersAndGroupsSearchProvider from ProcessRecord{611274c 32538:com.owncloud.android.beta/u0a72} (pid=32538, uid=10072) that is not exported from uid 10136
````
````
12-25 10:35:25.914 W/SuggestionsAdapter(32538): java.lang.SecurityException: Permission Denial: opening provider com.owncloud.android.providers.UsersAndGroupsSearchProvider from ProcessRecord{611274c 32538:com.owncloud.android.beta/u0a72} (pid=32538, uid=10072) that is not exported from uid 10136
````

**Comment by tobiasKaminsky** [comment]
Strange, it tries to use com.owncloud.android.providers.UsersAndGroupsSearchProvider where it should use org.owncloud.beta.android.providers.UsersAndGroupsSearchProvider.

Do you have installed the official version in parallel?

**Comment by cwmke** [comment]
I do.  I installed it again after I discovered the problem to test and see if the problem was likely with the Nexus or with the beta version.

**Comment by elpraga** [comment]
The problem persists in latest beta 20160103 sharing with users does not work. 

Can someone check if the bug is present when only beta is installed?

**Comment by AndyScherzinger** [comment]
Tested it with just the latest beta installed and I can confirm @elpraga's report. Nothing pops up during search. The only thing I see in the logcat is:
```01-07 19:11:15.717 8567-14693/com.owncloud.android.beta E/ActivityThread: Failed to find provider info for com.owncloud.android.providers.UsersAndGroupsSearchProvider```

And it can't work... I think I might have an idea about the problem...

public static final String AUTHORITY = UsersAndGroupsSearchProvider.class.getCanonicalName();
...
mUriMatcher.addURI(AUTHORITY, SearchManager.SUGGEST_URI_PATH_QUERY + "/*", SEARCH);
...
 int match = mUriMatcher.match(uri);

I did change the xml config though:
<searchable xmlns:android="http://schemas.android.com/apk/res/android"
    android:label="@string/app_name"
    android:hint="@string/search_users_and_groups_hint"
    android:searchSuggestAuthority="@string/authoritySearch"
    android:searchSuggestIntentAction="com.owncloud.android.providers.UsersAndGroupsSearchProvider.action.SHARE_WITH"
    android:searchSuggestThreshold="1" >
</searchable>        switch (match) {
            case SEARCH:
                return searchForUsersOrGroups(uri);

            default:
                return null;
        }

So the (at least) first problem is that the matcher won't match and as such the search won't be triggered:

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky I just committed a fix for beta (namespacing stuff basically...) - see https://github.com/owncloud/android/commit/313756372a1d66cd3d18d55cc9ab957a414afcaa

**Comment by AndyScherzinger** [comment]
Na... to early... atm just the search works and clicking on an item doesn't do anything...

**Comment by AndyScherzinger** [comment]
Just found the last piece. Now it is working again on the beta branch

**Comment by tobiasKaminsky** [comment]
:+1: 
I will create a new beta tomorrow!

**Comment by AndyScherzinger** [comment]
Sweet! :+1: I already bumped the version to tomorrow so you don't have to :smiley:

**Comment by tobiasKaminsky** [comment]
@cwmke is this working for you?

**Comment by cwmke** [comment]
It is.  Sorry, I should have commented before closing.  It's fully functional.  I noticed an additional issue that I created a new issue for #1419.  Thanks for all of your help and the quick turn around.
